src/components/FeatureViews/Scheduler/EditSessionView.js

```diff
@@ -1,359 +1,660 @@
+          variables: {
+            courseCategoryId: course_id,
+            courseId: course_id,
+            courseTitle: title,
+            courseInstructor: instructor.value,
+            room: room,
+          },
+        });
+      }
+    }
+    history.push(
+      `/scheduler/view-session/${course_id}/${session_id}/${sessionFields.instructor.value}`
+    );
+  };
+
+  if (loading || categoriesLoading || instructorsLoading) {
+    return <Loading />;
+  }
+
+  if (error || categoriesError || instructorsError) {
+    return <Typography>There's been an error!</Typography>;
+  }
+
+  const categoriesList = categoriesData.courseCategories.map(
+    ({ id, name }) => ({
+      value: id,
+      label: name,
+    })
+  );
+
+  const instructorList = instructorsData.instructors.map((instructor) => ({
+    value: instructor.user.id,
+    label: `${fullName(instructor.user)} - ${instructor.user.email}`,
+  }));
+
+  const courseDurationOptions = [1, 1.5, 2, 0.5];
+
+  const course = data.session.course;
+
+  const session = data.session;
+
+  return (
+    <Grid container className="main-session-view">
+      <BackgroundPaper
+        elevation={2}
+        className="paper session"
+        mt="2em"
+        style={{ width: "100%" }}
+      >
+        <Grid className="session-button" item>
+          <BackButton />
+        </Grid>
+        <Grid className="session-view" container direction="row" spacing={2}>
+          <Grid item sm={12}>
+            <TextField
+              fullWidth
+              onChange={handleTextChange("title")}
+              value={sessionFields.title}
+            />
+          </Grid>
+          <Grid
+            align="left"
+            className="session-view-details"
+            container
+            spacing={2}
+            xs={6}
+          >
+            {location.state.allOrCurrent === "all" && (
+              <Grid item xs={6}>
+                <Typography variant="h5"> Subject </Typography>
+                <SearchSelect
+                  className="search-options"
+                  isClearable
+                  onChange={handleCategoryChange}
+                  options={categoriesList}
+                  placeholder="Choose a Category"
+                  value={sessionFields.category}
+                />
+              </Grid>
+            )}
+
+            <Grid item xs={6}>
+              <Typography variant="h5"> Room</Typography>
+              <TextField
+                onChange={handleTextChange("room")}
+                value={sessionFields.room}
+              />
+            </Grid>
+
+            <Grid item xs={12}>
+              <Typography variant="h5"> Instructor </Typography>
+              <SearchSelect
+                onChange={handleInstructorChange}
+                options={instructorList}
+                placeholder="Choose an Instructor"
+                value={sessionFields.instructor}
+              />
+              {location.state.allOrCurrent === "current" && (
```

Comment:
make a wrapper component that helps make this more readable. For example:
```
<EditCurrentSessionField>
    <FormControl>
         <InputLabel> Is Instructor confirmed?</InputLabel>
    </FormControl>
</EditCurrentSessionField>
```
so the `<EditCurrentSessionField>` would look something like
```
const EditCurrentSessionField = ({children}) => {
    const location = useLocation();
    if(location.state.allOrCurrent)
          return children;
    else
           return;
}
```

Reply:
Component made and implemented!